Header header

uint64[]   new_indices     # new_indices[0] == 3 => the 0th element becomes the third